x/evm/types/utils.go

```diff
@@ -105,11 +107,55 @@ func DecodeMsgLogsFromEvents(in []byte, msgIndex int, blockNumber uint64) ([]*et
 		return nil, fmt.Errorf("invalid message index: %d", msgIndex)
 	}
 
-	return logsFromTxResponse(nil, txResponses[msgIndex], blockNumber), nil
+	logs, err := logsFromTxResponse(nil, txResponses[msgIndex], blockNumber), nil
+	if err != nil {
+		return nil, err
+	}
+
+	if len(logs) == 0 {
+		logs, err = TxLogsFromEvents(events, msgIndex)
```

Comment:
need to bear in mind that this adds some overheads to new transactions that don't emit any logs as well.

Reply:
yea, not sure any other way to differentiate this